README.md
leaflet_maps
============

TYPO3 Extension using CloudMade Leaflet lightweight open-source JavaScript library

Do not use, it´s experimental state.

ChangeLog
2012-07-03 Peter Pietzschmann  <peter@pietzschmann.de>

	* initial code generated with kickstarter


tca.php
<?php
if (!defined('TYPO3_MODE')) {
	die ('Access denied.');
}

$TCA['tx_leafletmaps_markers'] = array (
	'ctrl' => $TCA['tx_leafletmaps_markers']['ctrl'],
	'interface' => array (
		'showRecordFieldList' => 'hidden,longitude,latitude,popuptext'
	),
	'feInterface' => $TCA['tx_leafletmaps_markers']['feInterface'],
	'columns' => array (
		'hidden' => array (		
			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.hidden',
			'config'  => array (
				'type'    => 'check',
				'default' => '0'
			)
		),
		'longitude' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:leaflet_maps/locallang_db.xml:tx_leafletmaps_markers.longitude',		
			'config' => array (
				'type' => 'input',	
				'size' => '30',	
				'eval' => 'trim',
			)
		),
		'latitude' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:leaflet_maps/locallang_db.xml:tx_leafletmaps_markers.latitude',		
			'config' => array (
				'type' => 'input',	
				'size' => '30',	
				'eval' => 'trim',
			)
		),
		'popuptext' => array (		
			'exclude' => 1,		
			'label' => 'LLL:EXT:leaflet_maps/locallang_db.xml:tx_leafletmaps_markers.popuptext',		
			'config' => array (
				'type' => 'text',
				'cols' => '30',
				'rows' => '5',
			)
		),
	),
	'types' => array (
		'0' => array('showitem' => 'hidden;;1;;1-1-1, longitude, latitude, popuptext;;;richtext[]:rte_transform[mode=ts_css|imgpath=uploads/tx_leafletmaps/rte/]')
	),
	'palettes' => array (
		'1' => array('showitem' => '')
	)
);
?>

ext_tables.php
<?php
if (!defined('TYPO3_MODE')) {
	die ('Access denied.');
}

t3lib_extMgm::addToInsertRecords('tx_leafletmaps_markers');

$TCA['tx_leafletmaps_markers'] = array (
	'ctrl' => array (
		'title'     => 'LLL:EXT:leaflet_maps/locallang_db.xml:tx_leafletmaps_markers',		
		'label'     => 'uid',	
		'tstamp'    => 'tstamp',
		'crdate'    => 'crdate',
		'cruser_id' => 'cruser_id',
		'default_sortby' => 'ORDER BY crdate',	
		'delete' => 'deleted',	
		'enablecolumns' => array (		
			'disabled' => 'hidden',
		),
		'dynamicConfigFile' => t3lib_extMgm::extPath($_EXTKEY).'tca.php',
		'iconfile'          => t3lib_extMgm::extRelPath($_EXTKEY).'icon_tx_leafletmaps_markers.gif',
	),
);


t3lib_div::loadTCA('tt_content');
$TCA['tt_content']['types']['list']['subtypes_excludelist'][$_EXTKEY.'_pi1']='layout,select_key,pages';


t3lib_extMgm::addPlugin(array(
	'LLL:EXT:leaflet_maps/locallang_db.xml:tt_content.list_type_pi1',
	$_EXTKEY . '_pi1',
	t3lib_extMgm::extRelPath($_EXTKEY) . 'ext_icon.gif'
),'list_type');
?>

ext_localconf.php
<?php
if (!defined('TYPO3_MODE')) {
	die ('Access denied.');
}
t3lib_extMgm::addUserTSConfig('
	options.saveDocNew.tx_leafletmaps_markers=1
');
t3lib_extMgm::addPageTSConfig('

	# ***************************************************************************************
	# CONFIGURATION of RTE in table "tx_leafletmaps_markers", field "popuptext"
	# ***************************************************************************************
RTE.config.tx_leafletmaps_markers.popuptext {
  hidePStyleItems = H1, H4, H5, H6
  proc.exitHTMLparser_db=1
  proc.exitHTMLparser_db {
    keepNonMatchedTags=1
    tags.font.allowedAttribs= color
    tags.font.rmTagIfNoAttrib = 1
    tags.font.nesting = global
  }
}
');

t3lib_extMgm::addPItoST43($_EXTKEY, 'pi1/class.tx_leafletmaps_pi1.php', '_pi1', 'list_type', 1);
?>

pi1/class.tx_leafletmaps_pi1.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2012 Peter Pietzschmann <peter@pietzschmann.de>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 * Hint: use extdeveval to insert/update function index above.
 */

require_once(PATH_tslib.'class.tslib_pibase.php');


/**
 * Plugin 'Show leaflet map' for the 'leaflet_maps' extension.
 *
 * @author	Peter Pietzschmann <peter@pietzschmann.de>
 * @package	TYPO3
 * @subpackage	tx_leafletmaps
 */
class tx_leafletmaps_pi1 extends tslib_pibase {
	var $prefixId      = 'tx_leafletmaps_pi1';		// Same as class name
	var $scriptRelPath = 'pi1/class.tx_leafletmaps_pi1.php';	// Path to this script relative to the extension dir.
	var $extKey        = 'leaflet_maps';	// The extension key.
	var $pi_checkCHash = true;
	
	/**
	 * The main method of the PlugIn
	 *
	 * @param	string		$content: The PlugIn content
	 * @param	array		$conf: The PlugIn configuration
	 * @return	The content that is displayed on the website
	 */
	function main($content, $conf) {
		$this->conf = $conf;
		$this->pi_setPiVarDefaults();
		$this->pi_loadLL();
		
		//$this->pi_initPIflexForm();
		
		$this->extConf = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf'][$this->extKey]);
	/*
		$content='
			<strong>This is a few paragraphs:</strong><br />
			<p>This is line 1</p>
			<p>This is line 2</p>
	
			<h3>This is a form:</h3>
			<form action="'.$this->pi_getPageLink($GLOBALS['TSFE']->id).'" method="POST">
				<input type="text" name="'.$this->prefixId.'[input_field]" value="'.htmlspecialchars($this->piVars['input_field']).'">
				<input type="submit" name="'.$this->prefixId.'[submit_button]" value="'.htmlspecialchars($this->pi_getLL('submit_button_label')).'">
			</form>
			<br />
			<p>You can click here to '.$this->pi_linkToPage('get to this page again',$GLOBALS['TSFE']->id).'</p>
		';
		*/
		
		// http://leaflet.cloudmade.com/reference.html
	
		
		$GLOBALS['TSFE']->additionalHeaderData[$this->prefixId] = '
			<link rel="stylesheet" href="'.$GLOBALS["TSFE"]->absRefPrefix.'typo3conf/ext/leaflet_maps/dist/leaflet.css" />
			<!--[if lte IE 8]>
			<link rel="stylesheet" href="'.$GLOBALS["TSFE"]->absRefPrefix.'typo3conf/ext/leaflet_maps/dist/leaflet.ie.css" />
			<![endif]-->
		';
		
		
		$content = "<div id=\"map\" style=\"height: 400px\"></div>";
		
		$content .= '<script src="'.$GLOBALS["TSFE"]->absRefPrefix.'typo3conf/ext/leaflet_maps/dist/leaflet.js"></script>';

	
		$content .= "<script>";
		
		$content .= "
		var map = new L.Map('map');

		var cloudmadeUrl = 'http://{s}.tile.cloudmade.com/".$this->extConf['api_key']."/997/256/{z}/{x}/{y}.png',
			cloudmadeAttribution = 'Map data &copy; 2011 OpenStreetMap contributors, Imagery &copy; 2011 CloudMade',
			cloudmade = new L.TileLayer(cloudmadeUrl, {maxZoom: 18, attribution: cloudmadeAttribution});

		map.setView(new L.LatLng(51.505, -0.09), 13).addLayer(cloudmade);


		";
		
		$content .= $this->buildMarkers ();

		$content .= $this->buildPolylines ();

		$content .= $this->buildLayersControl ();
		
		$content .= $this->buildClickEvent ();
		
		$content .= "</script>";
		
		#$GLOBALS['TSFE']->additionalJavaScript[$this->extKey] = $js;
	
		return $this->pi_wrapInBaseClass($content);
	}
	
	
	function buildMarkers () {
		
		return "
				// http://leaflet.cloudmade.com/examples/custom-icons.html
		var markerLocation = new L.LatLng(51.5, -0.09),
			marker = new L.Marker(markerLocation);

		map.addLayer(marker);
		marker.bindPopup('<b>Hello world!</b><br />I am a popup.').openPopup();
		";
		
	}
	
	
	function buildPolylines () {
		
		return "		// 
		var circleLocation = new L.LatLng(51.508, -0.11),
			circleOptions = {color: '#f03', opacity: 0.7},
			circle = new L.Circle(circleLocation, 500, circleOptions);

		circle.bindPopup('I am a circle.');
		map.addLayer(circle);


		var p1 = new L.LatLng(51.509, -0.08),
			p2 = new L.LatLng(51.503, -0.06),
			p3 = new L.LatLng(51.51, -0.047),
			polygonPoints = [p1, p2, p3],
			polygon = new L.Polygon(polygonPoints);

		polygon.bindPopup('I am a polygon.');
		map.addLayer(polygon);";
	}

	
	function buildLayersControl () {
		
		return "
		// http://leaflet.cloudmade.com/examples/layers-control.html
		var baseLayers = {
			'CloudMade': cloudmade,
			//'OpenStreetMap': osm
		};

		var overlays = {
			'Marker': marker,
			//'Roads': roadsLayer
		};

		layersControl = new L.Control.Layers(baseLayers, overlays);

		map.addControl(layersControl);

		";
		
	}

	function buildClickEvent () {
		return "
		map.on('click', onMapClick);

		var popup = new L.Popup();

		function onMapClick(e) {
			var latlngStr = '(' + e.latlng.lat.toFixed(3) + ', ' + e.latlng.lng.toFixed(3) + ')';

			popup.setLatLng(e.latlng);
			popup.setContent('You clicked the map at ' + latlngStr);
			map.openPopup(popup);
		}";
		
	}
	
	
}



if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/leaflet_maps/pi1/class.tx_leafletmaps_pi1.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/leaflet_maps/pi1/class.tx_leafletmaps_pi1.php']);
}

?>

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "leaflet_maps".
#
# Auto generated 03-07-2012 23:44
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'Show Leaflet Maps',
	'description' => 'Show Leaflet Maps',
	'category' => 'plugin',
	'author' => 'Peter Pietzschmann',
	'author_email' => 'peter@pietzschmann.de',
	'shy' => '',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'beta',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => 'uploads/tx_leafletmaps/rte/',
	'modify_tables' => '',
	'clearCacheOnLoad' => 0,
	'lockType' => '',
	'author_company' => '',
	'version' => '0.0.0',
	'constraints' => array(
		'depends' => array(
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:13:{s:9:"ChangeLog";s:4:"54ff";s:10:"README.txt";s:4:"ee2d";s:12:"ext_icon.gif";s:4:"1bdc";s:17:"ext_localconf.php";s:4:"fa0f";s:14:"ext_tables.php";s:4:"2e16";s:14:"ext_tables.sql";s:4:"7bfc";s:31:"icon_tx_leafletmaps_markers.gif";s:4:"475a";s:16:"locallang_db.xml";s:4:"e463";s:7:"tca.php";s:4:"7a7a";s:19:"doc/wizard_form.dat";s:4:"bc03";s:20:"doc/wizard_form.html";s:4:"c008";s:32:"pi1/class.tx_leafletmaps_pi1.php";s:4:"069f";s:17:"pi1/locallang.xml";s:4:"134b";}',
);

?>